pkg/apis/v1beta1/nodeclaim.go

```diff
@@ -138,13 +138,13 @@ type KubeletConfiguration struct {
 
 type NodeClassReference struct {
 	// Kind of the referent; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"
-	// +optional
+	// +required
```

Comment:
We can't just make these fields required outright since this will break users if we change this all-of-a-sudden. We need a migration path to ensuring that these fields always exist. We should look into defaulting through OpenAPI as an example for the AWS cloudprovider implementation to ensure that users aren't broken.